### PR TITLE
Return error if object not found when deleting

### PR DESCRIPTION
**Pull request by alexeykazakov**

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-500-1`

File: account/repository/user.go [hunk 1]
```diff
@@ -153,14 +153,17 @@ func (m *GormUserRepository) Delete(ctx context.Context, id uuid.UUID) error {
 
 	obj := User{ID: id}
 
-	err := m.db.Delete(&obj).Error
+	db := m.db.Delete(&obj)
 
-	if err != nil {
+	if db.Error != nil {
```
Comment (sbryzak) [inline account/repository/user.go]: Is this thread-safe?

Comment (sbryzak) [reply account/repository/user.go]: ah nevermind, I didn't notice that you're assigning to a new `db` variable :)

Comment (sbryzak) [reply account/repository/user.go]: Actually the `db` variable name is a little confusing, could we name it `result` instead, like in the example?

[http://gorm.io/docs/error_handling.html](http://gorm.io/docs/error_handling.html)

Comment (alexeykazakov) [reply account/repository/user.go]: Yeah.. 'result' would be a better name.

Comment (alexeykazakov) [reply account/repository/user.go]: Fixed in https://github.com/fabric8-services/fabric8-auth/pull/500/commits/6d5cd7acd5e170920b8f4b35097c72a8fd96bdf5

**Pull request merged**